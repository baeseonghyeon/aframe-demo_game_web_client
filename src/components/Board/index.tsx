import React, { Children } from 'react';
import styles from './Board.module.scss';
import cb from 'classnames/bind';

const cn = cb.bind(styles);

interface BoardProps {
  children?: React.ReactNode;
  title?: string;
  className?: string;
}

const Board = (props: BoardProps) => {
  const { children, title = '', className } = props;

  return (
    <div className={cn('container', className)}>
      {title && (
        <div className={cn('title__container')}>
          <h1 className={cn('title')}>{title}</h1>
        </div>
      )}
      <div className={cn('wrapper')}>{children}</div>
    </div>
  );
};

export default Board;
