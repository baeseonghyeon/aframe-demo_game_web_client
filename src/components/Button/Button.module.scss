@import '../../scss/global';

.container {
  z-index: 1;
  border-radius: 20px;
  border: 0px;
  cursor: pointer;
  font-size: 1.5rem;
  color: white;
  font-weight: bold;
  padding: 10px 20px;
  &:active {
    padding: 8px 18px;
  }
}

.theme {
  &--pink {
    background-color: $color-primary;
    box-shadow: 0px 8px 0px 0px $color-primary-variant;
    text-shadow: 1px 3px 4px $color-primary-variant;
    border-top: 3px solid #f64465;
    &:active,
    &:hover {
      background-color: #f64465;
      box-shadow: 0px 4px 0px 0px $color-primary-variant;
    }
  }
  &--yellow {
    background-color: $color-secondary;
    box-shadow: 0px 8px 0px 0px $color-secondary-variant;
    text-shadow: 1px 3px 4px $color-secondary-variant;
    border-top: 3px solid #faac0c;
    &:active,
    &:hover {
      background-color: #faac0b;
      box-shadow: 0px 4px 0px 0px $color-secondary-variant;
    }
  }
}

.shape {
  &--circle {
    border-radius: 100%;
  }
}
