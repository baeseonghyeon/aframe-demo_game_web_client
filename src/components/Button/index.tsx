import React from 'react';
import styles from './Button.module.scss';
import cb from 'classnames/bind';

const cn = cb.bind(styles);

interface ButtonProps {
  children?: React.ReactNode;
  className?: string;
  /**
   * 버튼 테마
   */
  theme?: 'pink' | 'yellow';
  /**
   * 버튼 곡선
   */
  shape?: 'circle';
}

const Button = (props: ButtonProps) => {
  const { theme = 'pink', shape, children, className } = props;

  return (
    <button
      className={cn(
        'container',
        `theme--${theme}`,
        `shape--${shape}`,
        className
      )}
    >
      {children}
    </button>
  );
};

export default Button;
