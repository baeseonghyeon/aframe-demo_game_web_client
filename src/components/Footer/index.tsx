import React from 'react';
import styles from './Footer.module.scss';
import cb from 'classnames/bind';

const cn = cb.bind(styles);

const Footer = () => {
  return <div className={cn('container')}>Â© 2021 A-Frame Demo Game</div>;
};

export default Footer;
