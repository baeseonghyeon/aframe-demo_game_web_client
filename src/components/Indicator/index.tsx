import React from 'react';
import styles from './Indicator.module.scss';
import cb from 'classnames/bind';

const cn = cb.bind(styles);

const Indicator = () => {
  return (
    <svg className={cn('indicator')} viewBox="0 0 100 100">
      <circle className={cn('circle')} cx="50" cy="50" r="45"></circle>
    </svg>
  );
};

export default Indicator;
