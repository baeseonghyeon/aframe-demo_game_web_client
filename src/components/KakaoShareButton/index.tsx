import React, { useEffect } from 'react';
import styles from './KakaoShareButton.module.scss';
import cb from 'classnames/bind';
import kakaoIcon from './assets/kakao.png';

const cn = cb.bind(styles);

interface KakaoShareButtonProps {
  children?: React.ReactNode;
  className?: string;
  onClick?: () => void;
}

const KakaoShareButton = (props: KakaoShareButtonProps) => {
  const { children, className, onClick } = props;

  useEffect(() => {
    createKakaoButton();
  }, []);

  const createKakaoButton = () => {
    if (window.Kakao) {
      const kakao = window.Kakao;
      // 중복 initialization 방지
      if (!kakao.isInitialized()) {
        kakao.init(process.env.REACT_APP_KAKAO_KEY);
      }

      kakao.Link.createCustomButton({
        container: '#kakao-link-btn',
        templateId: 56200
      });
    }
  };

  return (
    <div className={cn('container')}>
      {/* Kakao share button */}
      <button id="kakao-link-btn" className={cn('btn')}>
        <img className={cn('icon')} src={kakaoIcon} alt="kakao-share-icon" />
      </button>
    </div>
  );
};

export default KakaoShareButton;
