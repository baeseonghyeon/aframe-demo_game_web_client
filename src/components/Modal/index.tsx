import React from 'react';
import styles from './Modal.module.scss';
import cb from 'classnames/bind';

const cn = cb.bind(styles);

export interface ModalProps extends React.HtmlHTMLAttributes<HTMLDivElement> {
  /**
   * 컨테이너 클래스 네임
   */
  containerClassName?: string;
  /**
   * 모달 여부
   */
  isVisible: boolean;
  /**
   * 백드롭 여부
   */
  isBackdrop?: boolean;
  /**
   * 백드롭 스타일
   */
  backdropClassName?: string;
  /**
   * 백드롭 클릭 이벤트
   */
  onClickBackdrop?: () => void;
  /**
   * ref
   */
  ref?: any;
}

/**
 *  모달 컴포넌트 props : ({ isVisible, onClickBackdrop, backdropClassName })
 */
const Modal = (props: ModalProps) => {
  const {
    className,
    containerClassName,
    isVisible = false,
    isBackdrop = true,
    backdropClassName,
    onClickBackdrop,
    ref,
    ...rest
  } = props;

  // Visible Class
  const visibleClassNames = cn([
    containerClassName && containerClassName,
    isVisible ? `container--visible` : `container--none`,
    isVisible && isBackdrop ? `container--fill` : `container--fit`
  ]);

  // Modal Container Class
  const classNames = cn(props.className, 'wrapper');

  return (
    <div className={visibleClassNames} ref={ref} {...rest}>
      {/* Modal Content  */}
      <div className={classNames}>{isVisible && props.children}</div>
      {/* BackDrop Layout */}
      {isBackdrop && (
        <div
          className={cn('backdrop', backdropClassName)}
          onClick={onClickBackdrop}
        ></div>
      )}
    </div>
  );
};

export default Modal;
