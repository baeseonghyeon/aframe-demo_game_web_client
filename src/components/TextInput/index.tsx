import React from 'react';
import styles from './TextInput.module.scss';
import cb from 'classnames/bind';

const cn = cb.bind(styles);

export interface TextInputProps
  extends React.HtmlHTMLAttributes<HTMLDivElement | HTMLInputElement> {
  className?: string;
  placeholder?: string;
  value: string;
  autoFocus?: boolean;
  onChange?: (e: any) => void;
}

const TextInput = (props: TextInputProps) => {
  const { placeholder, value, className, autoFocus, onChange, ...rest } = props;

  return (
    <input
      className={cn('container', className)}
      placeholder={placeholder}
      value={value}
      onChange={onChange}
      autoFocus={autoFocus}
      {...rest}
    />
  );
};

export default TextInput;
