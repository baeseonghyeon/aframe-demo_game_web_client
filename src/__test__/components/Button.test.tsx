import React from 'react';
import ReactDOM from 'react-dom';
import { default as Button } from '../../components/Button';
import { render } from '@testing-library/react';

describe('[Component] Button', () => {
  it('스냅샷 테스트 : default', () => {
    const rendered = render(<Button />);
    expect(rendered.container).toMatchSnapshot();
  });
  it('프로퍼티 테스트 : children', () => {
    const __BTN_LABEL_TEXT__ = 'button__' + new Date().getTime();
    const { getByText } = render(
      <Button>
        <p>{__BTN_LABEL_TEXT__}</p>
      </Button>
    );
    getByText(__BTN_LABEL_TEXT__);
  });
});
