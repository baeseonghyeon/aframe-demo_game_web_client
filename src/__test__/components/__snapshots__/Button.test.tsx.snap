// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Button 스냅샷 테스트 : default 1`] = `
<div>
  <button
    class="container theme--pink shape--undefined"
  />
</div>
`;
